src/dune_pkg/rev_store.ml

```diff
@@ -724,12 +758,32 @@ let remote =
                ]
            ])
   in
-  fun t ~url ->
+  fun t ~url ~url_loc ->
     let f url =
       let command = [ "ls-remote"; url ] in
       let refs =
         Fiber_lazy.create (fun () ->
-          let+ hits = run_capture_lines t ~display:!Dune_engine.Clflags.display command in
+          let+ hits =
+            run_capture_lines t ~display:!Dune_engine.Clflags.display command
+            >>| function
+            | Ok lines -> lines
+            | Error git_error ->
+              (match git_error.exit_code with
+               | 128 ->
+                 let command_string = String.concat ~sep:" " command in
```

Comment:
This will generate an invalid command if any of the items in `command` have a space (which I guess a URL could have, unless it is URL-encoded). Maybe better always quote the second item to be on the safe side.